packages/ncids-js/src/components/nci-header/utils/search.ts

```diff
@@ -0,0 +1,224 @@
+import { FocusTrap } from './focustrap';
+
+/**
+ * NCI Search
+ *
+ *
+ * Initialize the Search component:
+ * ```
+ * Search.element.create(HTMLElement);
+ * ```
+ */
+export class Search {
+	/** The parent DOM element for overlay placement. */
+	protected element: HTMLElement;
+	/** The parent DOM that contains search form. */
+	protected search!: HTMLElement;
+	/** Search Input Box  */
+	protected searchField!: HTMLInputElement;
+	/** Focus Trap class for mobile search*/
+	private focusTrap: FocusTrap;
+	/** Mobile Search Active  */
+	protected mobileActive!: boolean;
+	/** Callback for canceling search.  */
+	private resizeListener: EventListener = (event: Event) =>
+		this.handleResize(<Event>event);
+	/** Callback for canceling search.  */
+	private cancelListener: EventListener = (event: Event) =>
+		this.handleCancel(<Event>event);
+	/** Callback activating search overlay.  */
+	private searchListener: EventListener = (event: Event) =>
+		this.handleSearch(<Event>event);
+
+	/**
+	 * Sets component variables and initializes component.
+	 *
+	 * @param {HTMLElement} element parent DOM where search is contained.
+	 * @public
+	 */
+	public constructor(element: HTMLElement) {
+		this.element = element;
+		// container for the search input box and buttons
+		this.search = <HTMLInputElement>(
+			element.querySelector('.nci-header-nav__secondary')
+		);
+		// input textbox element
+		this.searchField = <HTMLInputElement>(
+			this.element.querySelector('#nci-header-search__field')
+		);
+
+		this.focusTrap = new FocusTrap(this.search);
+
+		// check screen size to activate menu functionality for mobile
+		const header = <HTMLElement>document.getElementById('nci-header');
+		const screenWidth = header?.clientWidth;
+		this.mobileActive = screenWidth > 900 ? true : false;
+
+		this.initialize();
+	}
+
+	/**
+	 * Sets up component by initializing.
+	 * @private
+	 */
+	private initialize(): void {
+		this.searchField.addEventListener('focus', this.searchListener, false);
+		window.addEventListener('resize', this.resizeListener, false);
+	}
+
+	/**
+	 * Removes all events from searchbox
+	 * @public
+	 */
+	public unregister(): void {
+		this.searchField.removeEventListener('focus', this.searchListener, false);
+		window.removeEventListener('resize', this.resizeListener, false);
+	}
+
+	/**
+	 * Click handler for activating the overlays
+	 *
+	 * @param {Event} event Event from document event handler.
+	 *
+	 * @private
+	 */
+	private handleResize(event: Event): void {
+		event.preventDefault();
```

Comment:
This code is not being tested and it should. Also there are bugs with the resize that keep the overlay open, but hidden. On resize from mobile to desktop the overlay should be closed. @olitharp-nci is putting this as a bug for the backlog post MVP as it is an edge case. (But maintaining the state the way we are doing it is klunky and may end up biting us further down the road.)